src/main/java/com/javachallengers/simpson/service/SimpsonCharacterService.java

```diff
@@ -30,17 +33,58 @@ public SimpsonCharacterService(SimpsonCharacterRepository simpsonCharacterReposi
 	public SimpsonCharacter createNewCharacter(SimpsonCharacterRequestDTO simpsonCharacterRequestDTO) throws SimpsonCharacterException {
 		String name = simpsonCharacterRequestDTO.getName();
 		String surname = simpsonCharacterRequestDTO.getSurname();
-		
 		Optional<SimpsonCharacter> simpsonCharacter = simpsonCharacterRepository.findByNameAndSurname(name, surname);
 		if(simpsonCharacter.isPresent()) {
-			throw new SimpsonCharacterException("Character already exists with name " + name + " and surname " + surname);
+			throw new SimpsonCharacterException("Another character already exists with the same name " + name + " and surname " + surname);
 		}
 		
 		LocalDate birthDate = simpsonCharacterRequestDTO.getBirthDate();
 		String city = simpsonCharacterRequestDTO.getCity();
 		String country = simpsonCharacterRequestDTO.getCountry();
-		SimpsonCharacter newCharacter = new SimpsonCharacter(name, surname, birthDate, city, country);
 		
+		SimpsonCharacter newCharacter = SimpsonCharacter.builder()
+				.name(name)
+				.surname(surname)
+				.birthDate(birthDate)
+				.city(city)
+				.country(country)
+				.build();
+				
 		return simpsonCharacterRepository.save(newCharacter);
 	}
+
+	public void updateCharacter(String id, SimpsonCharacterRequestDTO data) throws SimpsonCharacterException {
+		Optional<SimpsonCharacter> currentCharacter = simpsonCharacterRepository.findById(id);
+		if(currentCharacter.isEmpty()) {
+			throw new SimpsonCharacterException("Character not found");
+		}
+		
+		Optional<SimpsonCharacter> anotherCharacter = simpsonCharacterRepository.findByNameAndSurname(data.getName(), data.getSurname());
+		if(anotherCharacter.isPresent() && !anotherCharacter.get().getId().equals(id)) {
+			throw new SimpsonCharacterException("Another character already exists with the same name " + data.getName() + " and surname " + data.getSurname());
+		}
+	
+		SimpsonCharacter character = currentCharacter.get();
```

Comment:
Can we create a converter class that encapsulates the object conversion?

Reply:
I didn't understand exactly. I refactored to use builder.